gorm-tools/src/main/groovy/gorm/tools/mango/jpql/PagedQuery.groovy

```diff
@@ -86,10 +86,14 @@ class PagedQuery {
             queryString = buildNamedParameterQueryFromGString((GString) queryString, params)
         }
 
+        Integer timeout = args.remove('timeout')
+
         List dataList = template.execute { Session session ->
             Query q = (Query) session.createQuery(queryString.toString())
             template.applySettings(q)
-
+            if(timeout) {
+                q.setTimeout(timeout)
+            }
```

Comment:
doesn't populateQueryArguments already do this?

Reply:
Fixed